pkg/scheduler/schedule_one.go

```diff
@@ -721,12 +723,9 @@ func findNodesThatPassExtenders(ctx context.Context, extenders []framework.Exten
 		}
 
 		for failedNodeName, failedMsg := range failedAndUnresolvableMap {
-			var aggregatedReasons []string
-			if _, found := statuses[failedNodeName]; found {
-				aggregatedReasons = statuses[failedNodeName].Reasons()
-			}
-			aggregatedReasons = append(aggregatedReasons, failedMsg)
-			statuses[failedNodeName] = framework.NewStatus(framework.UnschedulableAndUnresolvable, aggregatedReasons...)
+			status := statuses.Get(failedNodeName)
```

Comment:
What if `failedNodeName` was _not_ rejected by PreFilter/Filter extension point?
Doesn't `statuses.Get(failedNodeName)` return absent node status incorrectly?

Reply:
You're right, `absentNodesStatus` reasons should not be used here, so added 
`if status != statuses.AbsentNodesStatus()` there. 

I also modified the tests to catch this issue.